src/MANYBODY/pair_airebo.cpp

```diff
@@ -3094,72 +3092,54 @@ double PairAIREBO::gSpline(double costh, double Nij, int typei,
 double PairAIREBO::PijSpline(double NijC, double NijH, int typei, int typej,
                              double dN2[2])
 {
-  int x,y,i,done;
-  double Pij,coeffs[16];
-
-  for (i = 0; i < 16; i++) coeffs[i]=0.0;
+  int x,y;
+  double Pij;
 
   x = 0;
   y = 0;
   dN2[0] = 0.0;
   dN2[1] = 0.0;
-  done = 0;
+  Pij = 0.0;
 
-  // if inputs are out of bounds set them back to a point in bounds
+  if (typei == 1) return Pij;
 
-  if (typei == 0 && typej == 0) {
-    if (NijC < pCCdom[0][0]) NijC=pCCdom[0][0];
-    if (NijC > pCCdom[0][1]) NijC=pCCdom[0][1];
-    if (NijH < pCCdom[1][0]) NijH=pCCdom[1][0];
-    if (NijH > pCCdom[1][1]) NijH=pCCdom[1][1];
+  if (typej == 0) {
 
-    if (fabs(NijC-floor(NijC)) < TOL && fabs(NijH-floor(NijH)) < TOL) {
-      Pij = PCCf[(int) NijC][(int) NijH];
-      dN2[0] = PCCdfdx[(int) NijC][(int) NijH];
-      dN2[1] = PCCdfdy[(int) NijC][(int) NijH];
-      done = 1;
-    }
-    if (done == 0) {
-      x = (int) (floor(NijC));
-      y = (int) (floor(NijH));
-      for (i = 0; i<16; i++) coeffs[i] = pCC[x][y][i];
-      Pij = Spbicubic(NijC,NijH,coeffs,dN2);
-    }
-  }
-
-  // if inputs are out of bounds set them back to a point in bounds
+    // if inputs are out of bounds set them back to a point in bounds
 
-   if (typei == 0 && typej == 1){
-     if (NijC < pCHdom[0][0]) NijC=pCHdom[0][0];
-     if (NijC > pCHdom[0][1]) NijC=pCHdom[0][1];
-      if (NijH < pCHdom[1][0]) NijH=pCHdom[1][0];
-      if (NijH > pCHdom[1][1]) NijH=pCHdom[1][1];
+    if (NijC <  pCCdom[0][0]) NijC=pCCdom[0][0];
+    if (NijC >= pCCdom[0][1]) NijC=pCCdom[0][1]-TOL;
+    if (NijH <  pCCdom[1][0]) NijH=pCCdom[1][0];
+    if (NijH >= pCCdom[1][1]) NijH=pCCdom[1][1]-TOL;
+    x = (int) floor(NijC);
+    y = (int) floor(NijH);
 
     if (fabs(NijC-floor(NijC)) < TOL && fabs(NijH-floor(NijH)) < TOL) {
```

Comment:
This check will now fail for for integer values along the boundary, where it used to work.
These nodal checks are nice because they are very exact and also quite fast.